File: cv-builder/src/main/java/model/embedded/components/PersonalDetailsObject.java
package model.embedded.components;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import model.embedded.personal.LanguageObject;
import model.embedded.personal.LinkObject;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class PersonalDetailsObject {

    private String firstName;
    private String lastName;
    private String phone;
    private String email;
    private String city;
    private List<LanguageObject> languages;
    private List<LinkObject> links;
}

File: frontend/src/types/embedded/components/PersonalDetailsObject.ts
import type { LanguageObject } from "../personal/LanguageObject";
import type { LinkObject } from "../personal/LinkObject";

export interface PersonalDetailsObject {
    firstName: string;
    lastName: string;
    phone: string;
    email: string;
    city: string;
    languages: LanguageObject[];
    links: LinkObject[];
}